src/neuroconv/nwbconverter.py

```diff
@@ -158,10 +159,24 @@ def create_nwbfile(self, metadata: Optional[dict] = None, conversion_options: Op
 
     def add_to_nwbfile(self, nwbfile: NWBFile, metadata, conversion_options: Optional[dict] = None) -> None:
         conversion_options = conversion_options or dict()
-        for interface_name, data_interface in self.data_interface_objects.items():
-            data_interface.add_to_nwbfile(
-                nwbfile=nwbfile, metadata=metadata, **conversion_options.get(interface_name, dict())
-            )
+        for interface_key, data_interface in self.data_interface_objects.items():
```

Comment:
I think interface name is a better variable name, why change this?

Reply:
It was never the name of the interface, it was whatever key the individual assigned to the entries of the `data_interface_classes` (or autogenerated from a `ConverterPipe`)

Neither name is perfect though since it can be either an interface or a subconverter

So maybe just `key`? or `item_key`?

Reply:
Are you coming from the angle that a class name is **only**  class.__name__? If so, fine. I think that interface_key is OK if you want to make that distinction.  I had a more general that name is whatever we use to refer to the interface but this is less important. I would not go for generic key or item_key os the current is OK.

Reply:
I'm coming at the angle that is it a key in a dictionary and does not need to match or even correlate with `interface.__name__`

Reply:
In the GUIDE we refer to this as the more verbose `interface_or_subconverter_key`

Reply:
In my mind the name of an interface is not `interface.__name__`. I think that's  fine identification to have though so I think interface_key is OK.

I don't like the thing we did with the passing converter pipes as interfaces here tbh, the naming is wrong and I think it will create problems in the future.

Reply:
How else would we perform a single `run_conversion` call of both multi-probe SpikeGLX (using the Converter) and Phy sorting (single interface)?

Or multi-plane Bruker with PoseEstimation and ROI responses?

We made prolific use of this in the NWB GUIDE, where you can easily compose multi-stream converters with single-stream interfaces. In terms of their APIs, this issue with `conversion_options` is the only difference to reconcile

Reply:
Yeah, I did not realize you went that way until I started reading this.